addons/web/static/tests/core/position_hook_tests.js

```diff
@@ -812,3 +812,42 @@ QUnit.test(
     "reposition from right-end to top-end",
     getRepositionTest("right-end", "top-end", "w125 right")
 );
+
+function getFittingTest(position, style) {
+    return async (assert) => {
+        const TestComp = getTestComponent({
+            position,
+        });
+        await mount(TestComp, container);
+        Object.assign(document.querySelector("#reference").style, style);
+        const isVertical = ["bottom", "top"].includes(position.split("-")[0]);
+        const styleAttribute = isVertical ? "width" : "height";
+        triggerEvent(document, null, "scroll");
+        await nextTick();
+        assert.strictEqual(container.querySelector("#popper").style[styleAttribute], isVertical ? style.width : style.height);
+    }
+}
```

Comment:
```suggestion
function getFittingTest(position) {
    return async (assert) => {
        const TestComp = getTestComponent({ position });
        const [direction] = position.split("-");
        const isVertical = ["bottom", "top"].includes(direction);
        const styleAttribute = isVertical ? "width" : "height";
        await mount(TestComp, container);
        assert.strictEqual(container.querySelector("#popper").style[styleAttribute], "50px");
    }
}
```

or instead of computing the style attribute:
```js
function getFittingTest(position, styleAttribute) {
    return async (assert) => {
        const TestComp = getTestComponent({ position });
        await mount(TestComp, container);
        assert.strictEqual(container.querySelector("#popper").style[styleAttribute], "50px");
    }
}
```